tests/components/caldav/test_calendar.py

```diff
@@ -1140,3 +1161,115 @@ async def test_config_entry_supported_components(
+                    2025, 8, 6, 11, 0, tzinfo=zoneinfo.ZoneInfo(key="UTC")
+                ),
+                "dtstart": datetime.datetime(
+                    2025, 8, 6, 10, 0, tzinfo=zoneinfo.ZoneInfo(key="UTC")
+                ),
+                "location": "Test Location",
+                "summary": "Test Event",
+            },
+        ),
+        # Event with only required fields
+        (
+            {
+                "summary": "Required Only",
+                "start_date_time": "2025-08-07T09:00:00+00:00",
+                "end_date_time": "2025-08-07T10:00:00+00:00",
+            },
+            {
+                "summary": "Required Only",
+                "dtstart": datetime.datetime(
+                    2025, 8, 7, 9, 0, tzinfo=zoneinfo.ZoneInfo(key="UTC")
+                ),
+                "dtend": datetime.datetime(
+                    2025, 8, 7, 10, 0, tzinfo=zoneinfo.ZoneInfo(key="UTC")
+                ),
+            },
+        ),
+        # All-day event (date only)
+        (
+            {
+                "summary": "All Day Event",
+                "start_date": "2025-08-08",
+                "end_date": "2025-08-09",
+            },
+            {
+                "summary": "All Day Event",
+                "dtstart": datetime.date(2025, 8, 8),
+                "dtend": datetime.date(2025, 8, 9),
+            },
+        ),
+        # Event with different timezone
+        (
+            {
+                "summary": "Different TZ",
+                "start_date_time": "2025-08-07T09:00:00+02:00",
+                "end_date_time": "2025-08-07T10:00:00+02:00",
+            },
+            {
+                "summary": "Different TZ",
+                "dtstart": datetime.datetime(
+                    2025, 8, 7, 7, 0, tzinfo=zoneinfo.ZoneInfo(key="UTC")
+                ),
+                "dtend": datetime.datetime(
+                    2025, 8, 7, 8, 0, tzinfo=zoneinfo.ZoneInfo(key="UTC")
+                ),
+            },
+        ),
+        # Rrule is not supported in API (async_call) calls.
+    ],
+)
+async def test_add_vevent(
+    hass: HomeAssistant,
+    setup_platform_cb: Callable[[], Awaitable[None]],
+    calendars: list[Mock],
+    service_data: dict,
+    expected_ics_fields: dict,
+) -> None:
+    """Test adding an event to the calendar."""
+    await setup_platform_cb()
+    hass_time_zone = zoneinfo.ZoneInfo(hass.config.time_zone)
+
+    calendars[0].add_event = MagicMock(return_value=[])
+    await hass.services.async_call(
+        "calendar",
+        "create_event",
+        service_data,
+        target={"entity_id": TEST_ENTITY},
+        blocking=True,
+    )
+    await hass.async_block_till_done()
+
+    calendars[0].add_event.assert_called_once()
+    assert calendars[0].add_event.call_args
+    actual_fields = calendars[0].add_event.call_args[1]
+
+    for key in ("dtstart", "dtend"):
+        if isinstance(actual_fields[key], datetime.date):
```

Comment:
OK, we don't like adding conditionals like this in tests, but i see why you did it. I'll take back what i said and you can revert this back to how it was.

Reply:
Reverted :)